api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -2792,6 +2792,52 @@ public void copyNotebook_alreadyExists() {
         copyNotebookRequest);
   }
 
+  @Test(expected = BadRequestException.class)
+  public void testCopyNotebook_notAllowedBetweenTiers() {
+    final DbAccessTier controlledTier = TestMockFactory.createControlledTierForTests(accessTierDao);
+
+    DbCdrVersion controlledCdr =
+        TestMockFactory.createDefaultCdrVersion(cdrVersionDao, accessTierDao, 2);
+    controlledCdr.setName("2");
+    controlledCdr.setAccessTier(controlledTier);
+    controlledCdr = cdrVersionDao.save(controlledCdr);
+
+    Workspace fromWorkspace = createWorkspace();
+    fromWorkspace.setCdrVersionId(String.valueOf(controlledCdr.getCdrVersionId()));
+    fromWorkspace = workspacesController.createWorkspace(fromWorkspace).getBody();
+    String fromNotebookName = "origin";
+
+    stubGetWorkspace(
+        fromWorkspace.getNamespace(),
+        fromWorkspace.getName(),
+        LOGGED_IN_USER_EMAIL,
+        WorkspaceAccessLevel.OWNER);
+
+    Workspace toWorkspace = testMockFactory.createWorkspace("toWorkspaceNs", "toworkspace");
```

Comment:
nit: `toworkspace` -> `toWorkspace`

Reply:
heh.  soooo there seems to be a subtle bug in this test where you MUST name the workspace in all-lowercase.  I'm guessing it's confusing firecloudName with aouName somewhere (firecloudName is aouName.tolower() by default) but I did not take the time to track it down.

Worth a TODO anyway.

Reply:
ahh, got it. a TODO comment sounds good